Use futex(2) for the libpas lock slow path, like Linux does.

Index: Source/bmalloc/libpas/src/libpas/pas_lock.h
--- Source/bmalloc/libpas/src/libpas/pas_lock.h.orig
+++ Source/bmalloc/libpas/src/libpas/pas_lock.h
@@ -165,7 +165,7 @@ static inline void pas_lock_testing_assert_held(pas_lo
 
 PAS_END_EXTERN_C;
 
-#elif PAS_OS(LINUX) || PAS_OS(WINDOWS) || PAS_OS(FREEBSD)
+#elif PAS_OS(LINUX) || PAS_OS(WINDOWS) || PAS_OS(FREEBSD) || PAS_OS(OPENBSD)
 
 PAS_BEGIN_EXTERN_C;
 
